SonarQube.TeamBuild.PostProcessor/Program.cs

```diff
@@ -46,15 +47,12 @@ static int Main()
             {
                 return ErrorCode;
             }
-            ProjectInfoAnalysisResult result = InvokeSonarRunner(config, logger);
 
-            // Write summary report
-            if (settings.BuildEnvironment == BuildEnvironment.LegacyTeamBuild
-                && !TeamBuildSettings.SkipLegacyCodeCoverageProcessing)
-            {
-                UpdateTeamBuildSummary(config, result, logger);
-            }
+            ProjectInfoAnalysisResult result = InvokeSonarRunner(config, logger);
 
+            ISonarPropertyProvider propertyProvider = new FilePropertiesProvider(config.SonarRunnerPropertiesPath);
+            SummaryReportBuilder.GenerateReports(settings, config, result, propertyProvider, logger);
+            
```

Comment:
To make the summary logic more testable I am passing in a property provider that tests can easily mock.